src/Symfony/Component/HttpKernel/ControllerMetadata/Argument/ArgumentMetadata.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\ControllerMetadata\Argument;
+
+/**
+ * Responsible for storing metadata of an argument.
+ *
+ * @author Iltar van der Berg <kjarli@gmail.com>
+ */
+class ArgumentMetadata
+{
+    private $argumentName;
+    private $argumentType;
+    private $isVariadic;
+    private $hasDefaultValue;
+    private $defaultValue;
+
+    public function __construct($argumentName, $argumentType, $isVariadic, $hasDefaultValue, $defaultValue)
+    {
+        $this->argumentName = $argumentName;
+        $this->argumentType = $argumentType;
+        $this->isVariadic = $isVariadic;
+        $this->hasDefaultValue = $hasDefaultValue;
+        $this->defaultValue = $defaultValue;
+    }
+
+    /**
+     * Returns the name as given in PHP, $foo would yield "foo".
+     *
+     * @return string
+     */
+    public function getArgumentName()
```

Comment:
Good thing you bring that up, while using it I was already having second thoughts. I'll remove "argument" from both method names.